xarray/backends/netCDF4_.py

```diff
@@ -485,21 +496,64 @@ def encode_variable(self, variable):
         return variable
 
     def prepare_variable(
-        self, name, variable, check_encoding=False, unlimited_dims=None
+        self, name, variable: Variable, check_encoding=False, unlimited_dims=None
     ):
         _ensure_no_forward_slash_in_name(name)
-
-        datatype = _get_datatype(
-            variable, self.format, raise_on_invalid_encoding=check_encoding
-        )
         attrs = variable.attrs.copy()
-
         fill_value = attrs.pop("_FillValue", None)
-
         encoding = _extract_nc4_variable_encoding(
             variable, raise_on_invalid=check_encoding, unlimited_dims=unlimited_dims
         )
-
+        if (
+            encoding.get("enum")
+            and attrs.get("flag_meanings")
+            and attrs.get("flag_values")
+        ):
+            var_enum_dict = {
+                k: v for k, v in zip(attrs["flag_meanings"], attrs["flag_values"])
+            }
+            enum_name = encoding["enum"]
```

Comment:
@bzah pop enum here from the dict, that will fix your tests.
```suggestion
            enum_name = encoding.pop("enum")
```

Reply:
Yep, but I was wondering if the goal of #7551 was to restrict the keys of encoding to the arguments of netCDF4 `createVariable` or if it was still right to put my enum key here.

Reply:
AFAICT, #7551 was meant for the new encoding arguments, but I could misinterpret. Do we want encoding["enum"] as argument to createVariable?

Reply:
Yes, I see the issue, but it feels like those keys have to be removed by the appropriate encoding functions before reaching that point.